### PR TITLE
Remove automated NPM publishing, add custom workflow trigger to publish github releases

### PR DESCRIPTION
**Pull request by mmv08**
This PR:
- Is needed for https://github.com/gnosis/safe-apps-sdk/issues/269
- Removes automated NPM publishing from the action
- Adds custom `workflow_dispatch` trigger. If the action is triggered with it, it will publish releases on GitHub
The action loops through packages on the custom trigger and checks if there's a corresponding GitHub release with a name: `@package-name@version`. If it doesn't find one, it takes changes for this version from the changelog and creates a release pointing to a `version` tag.

**How the release flow will work with the updated action**:
The action still behaves the same for pre-release states, so everything stays the same: we create PRs with changesets, and when they're pushed to `development` or `main`, the action will generate PR with updated versions/changelogs. Finally, when we're ready to release, we merge this PR and then:
1. Someone with publishing access runs `npx changesets publish --otp=XXXXXX` on their computer. This command publishes releases and creates tags
2. Pushes tags to remote
3. Runs the action from the `Actions` tab on Github

I tested the action on my forked safe-apps-sdk repo:
https://github.com/mikheevm/safe-apps-sdk

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
     "@changesets/pre": "^1.0.4",
     "@changesets/read": "^0.4.6",
     "@manypkg/get-packages": "^1.1.1",
+    "@octokit/webhooks-definitions": "^3.67.3",
```
Comment (DaniSomoza) [inline package.json]: It seems that @octokit/webhooks-definitions is deprecated, right?
https://www.npmjs.com/package/@octokit/webhooks-definitions

Captura de pantalla 2022-01-20 a las 12 12 18

[They recommend here](https://github.com/octokit/webhooks/issues/447#issue-862139322) using one of the following 3 packages now, depending on what you need:

```
@octokit/webhooks-types - TypeScript definitions
@octokit/webhooks-schemas - JSON Schemas
@octokit/webhooks-examples - Examples for all webhook events
```

Comment (mmv08) [reply package.json]: removed it caballero

File: src/index.ts [hunk 2]
```diff
@@ -14,78 +15,44 @@ const getOptionalInput = (name: string) => core.getInput(name) || undefined;
...
 
-  let setupGitUser = core.getBooleanInput('setupGitUser');
+  let setupGitUser = core.getBooleanInput("setupGitUser");
 
   if (setupGitUser) {
     console.log("setting git user");
     await gitUtils.setupUser();
   }
 
+  console.log(core);
```
Comment (yagopv) [inline src/index.ts]: Is this intented?

Comment (mmv08) [reply src/index.ts]: nop, removed it, sorry

File: .gitignore [hunk 3]
```diff
@@ -3,4 +3,4 @@ node_modules
 .cache
 *.log
 
-dist/
+# dist/
```
Comment (yagopv) [inline .gitignore:6]: Do you want to leave it as a comment?

**Review by yagopv — APPROVED** [review]

File: .gitignore (hunk 3 above)
Comment (mmv08) [reply .gitignore:6]: yes, when the action is run, it uses files from the `dist` folder, and therefore, the folder has to be committed

**Pull request merged**